java/arcs/core/analysis/InformationFlow.kt

```diff
@@ -0,0 +1,327 @@
+    private val labels = ArrayList(labelsInClaims union labelsInChecks)
+
+    /** Index assigned to the labels for use in bitsets. */
+    private val labelIndices: Map<InformationFlowLabel, Int> = labels.mapIndexed {
+        index, label -> label to index
+    }.toMap()
+
+    override fun getInitialValues(graph: RecipeGraph): Map<RecipeGraph.Node, AccessPathLabels> {
+        // TODO(bgogul): This is where we get initial value from the mapped stores.
+        // TODO(bgogul): Ensure that ingress particles do not have input connections.
+        val nodeValues = mutableMapOf<RecipeGraph.Node, AccessPathLabels>()
+        ingresses.forEach { ingressSpec ->
+            getInitialValueForIngress(graph, ingressSpec)?.let {
+                (node, initialValues) -> nodeValues[node] = initialValues
+            }
+        }
+        log.debug { "Initial Values For Ingress Nodes:\n $nodeValues" }
+        return nodeValues.toMap()
+    }
+
+    /**
+     * Returns the initial abstract value to be used for the given [ingressSpec]. An [ingressSpec]
+     * is of the form "<particle-name>.<connection-name>".
+     */
+    private fun getInitialValueForIngress(
+        graph: RecipeGraph,
+        ingressSpec: String
+    ): Pair<RecipeGraph.Node, AccessPathLabels>? {
+        val specParts = ingressSpec.split(".")
+        require(specParts.size <= 2) { "Unsupported ingress specification." }
+        val particleName = specParts[0]
+        val connectionName = if (specParts.size == 2) specParts[1] else null
+        val particleNode = graph.nodes.asSequence()
+            .filterIsInstance<RecipeGraph.Node.Particle>()
+            .find { it.particle.spec.name == particleName }
+        val initialValues = mutableMapOf<AccessPath, InformationFlowLabels>()
+
+        val particle = particleNode?.particle ?: return null
+
+        // If a connection is specified, set it to empty labels.
+        // Otherwise, set all write connections to empty labels.
+        particle.handleConnections
+            .filter { connectionName?.equals(it.spec.name) ?: it.spec.isWrite() }
+            .forEach {
+                val emptyLabels = InformationFlowLabels(setOf(BitSet(labels.size)))
+                val accessPath = AccessPath(particle, it.spec)
+                initialValues[accessPath] = emptyLabels
+            }
+
+        // For read-write connections, apply any claims that are present. This is necessary as
+        // the claim on a read-write connection immediately manifests on the read connection.
+        particleClaims[particle]?.let { claims ->
+            initialValues.applyClaims(
+                claims.filterIsInstance<Claim.Assume>()
+                    .filter {
+                        val accessPathRoot =
+                            it.accessPath.root as? AccessPath.Root.HandleConnection
+                        accessPathRoot?.connectionSpec?.isReadWrite() == true
```

Comment:
Nit: you don't need the `== true`

Reply:
`== true` is needed as the expression is a nullable Boolean. See https://kotlinlang.org/docs/reference/idioms.html#consuming-a-nullable-boolean